src/utils.ts

```diff
@@ -4,7 +4,34 @@ import type {
   CommonBlockProps,
   FileBlockProps,
   FolderBlockProps,
-} from "@utils";
+} from "../utils/types";
+
+import { Endpoints, OctokitResponse, RequestParameters } from "@octokit/types";
+import { endpoint } from "@octokit/endpoint";
+
+type FilterConditionally<Source, Condition> = Pick<
+  Source,
+  { [K in keyof Source]: K extends Condition ? K : never }[keyof Source]
+>;
+type GetEndpoints = FilterConditionally<Endpoints, `GET ${string}`>;
+
+export async function onRequestGitHubEndpoint<
+  Endpoint extends keyof GetEndpoints,
+  EndpointParameters extends GetEndpoints[Endpoint]["parameters"]
+>(
+  route: Endpoint,
+  parameters?: EndpointParameters & RequestParameters
+): Promise<OctokitResponse<Endpoints[Endpoint]>["data"]["response"]["data"]> {
+  const params = endpoint<Endpoint, EndpointParameters>(route, parameters);
+
+  if (params.method !== "GET") {
+    throw new Error("Only GET requests are supported.");
+  }
+
+  return makeRequest("onRequestGitHubEndpoint", params);
```

Comment:
see comment at https://github.com/githubnext/blocks-sandbox/pull/20/files#r1039990153 about moving the `endpoint` call to `blocks-platform`